Sources/NostrSDK/NIP44v2Encrypting.swift

```diff
@@ -0,0 +1,299 @@
+//
+//  NIP44v2Encrypting.swift
+//
+//
+//  Created by Terry Yiu on 3/16/24.
+//
+
+import Foundation
+import Clibsodium
+import CryptoKit
+import secp256k1
+
+public enum NIP44v2EncryptingError: Error {
+    case aadLengthInvalid(Int)
+    case base64EncodingFailed
+    case chaCha20DecryptionFailed
+    case chaCha20EncryptionFailed
+    case conversationKeyLengthInvalid(Int)
+    case dataSizeInvalid(Int)
+    case macInvalid
+    case nonceLengthInvalid(Int)
+    case paddingInvalid
+    case payloadSizeInvalid(Int)
+    case plaintextLengthInvalid(Int)
+    case privateKeyInvalid
+    case publicKeyInvalid
+    case sharedSecretComputationFailed
+    case unknownVersion(Int? = nil)
+    case unpaddedLengthInvalid(Int)
+    case utf8EncodingFailed
+}
+
+internal struct DecodedPayload {
+    let nonce: Data
+    let ciphertext: Data
+    let mac: Data
+}
+
+internal struct MessageKeys {
```

Comment:
Ah, and it looks like I can't declare it fileprivate because it needs to be used by the internal `decodePayload` function in the `NIP44v2Encrypting` extension. I'll just remove the access control modifier to have it be internal by default.